.github/workflows/main.yml

```diff
@@ -140,146 +198,27 @@ jobs:
-          -DCMAKE_BUILD_TYPE=RelWithDebInfo -DHDF5_ENABLE_THREADSAFE=OFF \
-          -DCMAKE_INSTALL_PREFIX=${{github.workspace}}/hdf5install \
-          ..
-          make 
-          make install
-        shell: bash
-        working-directory: ${{github.workspace}}/hdf5
-
-      - name: CMake Configure REST VOL
-        run: |
-          mkdir ./build
-          cd ./build
-          CFLAGS="-D_POSIX_C_SOURCE=200809L" cmake -G "Unix Makefiles" -DHDF5_ROOT=${{github.workspace}}/hdf5install \
-          -DCMAKE_INSTALL_PREFIX=${{github.workspace}}/vol-rest/install \
-          ..
-        shell: bash
-        working-directory: ${{github.workspace}}/vol-rest
-
-      - name: Build + Install REST VOL
-        run: |
-          make
-          make install
-        shell: bash
-        working-directory: ${{github.workspace}}/vol-rest/build
-
-      - uses: actions/checkout@v4
-        with:
-          repository: HDFGroup/hsds
-          path: ${{github.workspace}}/hsds
-          
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
-          
-      - name: Install HSDS dependencies
-        shell: bash
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install pytest
-          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-
-      - name: Install HSDS package
-        shell: bash
-        run: |
-          cd ${{github.workspace}}/hsds
-          pip install -e .
-          
-      - name: Run HSDS unit tests
-        shell: bash
-        run: |
-          cd ${{github.workspace}}/hsds
-          pytest
-
-      - name: Install valgrind
-        run: |
-          sudo apt update
-          sudo apt install valgrind
-        working-directory: ${{ github.workspace }}
-
-        # Requests 2.32.0 breaks requests-unixsocket, used by HSDS for socket connections
-      - name: Fix requests version
+      - name: Test REST VOL (CMake)
+        if: matrix.build_system == 'cmmake'
+        working-directory: ${{github.workspace}}/vol-rest/build/
         run: |
-          pip install requests==2.31.0
-
-      - name: Start HSDS
-        if: ${{ matrix.endpoint != 'http://127.0.0.1:5101'}}
-        run: | 
-            cd ${{github.workspace}}/hsds
-            mkdir ${{github.workspace}}/hsdsdata &&
-            mkdir ${{github.workspace}}/hsdsdata/hsdstest &&
-            cp admin/config/groups.default admin/config/groups.txt &&
-            cp admin/config/passwd.default admin/config/passwd.txt &&
-            cp admin/config/groups.default admin/config/groups.txt &&
-            cp admin/config/passwd.default admin/config/passwd.txt
-            ROOT_DIR=${{github.workspace}}/hsdadata ./runall.sh --no-docker 1 &
-            sleep 10
-        working-directory: ${{github.workspace}}/hsds
+              valgrind --leak-check=full -s ctest -R "test_rest_vol" -VV
```

Comment:
There's no real reason to use valgrind here since `ctest` will be wrapping running the tests. We also shouldn't generally use valgrind in the CI unless there's a specific reason